Lib/test/libregrtest/main.py

```diff
@@ -331,6 +333,19 @@ def run_tests_sequentially(self, runtests) -> None:
 
             result = self.run_test(test_name, runtests, tracer)
 
+            # Unload the newly imported modules (best effort finalization)
+            new_modules = [module for module in sys.modules
+                           if module not in save_modules and
+                                module.startswith("test.")]
+            for module in new_modules:
+                sys.modules.pop(module, None)
+                # Remove the attribute of the parent module.
+                parent, _, name = module.rpartition('.')
```

Comment:
Maybe raise an error if the parent itself has a parent other than `test.`. Currently, it should not happen. But it may happen in the future, if tests are reorganized.

Reply:
What do you mean? The parent should be "test" or start with "test.", because the module starts with "test.". If the parent starts with "test." it will be removed as well. If it was already removed before children, the code ignores KeyError.

Reply:
If a test imports `test.test_a.test_b.test_c`, does this code unloads `test.test_a`, `test.test_a.test_b` and `test.test_a.test_b.test_c`?